client/src/components/admin/billing/Billing.js

```diff
@@ -99,191 +92,109 @@ const styles = theme => ({
-          { date: '5/3/2018', amount: '$440.00' },
-        ],
-      },
-    ],
+    hasStripeID: false,
+    fetchingStripeID: true,
   };
   componentDidMount() {
     console.log('props', this.props);
-    if (this.props.location.search) {
-      console.log('BINGO', this.props.location.search.substring(23));
-      let computedCode = this.props.location.search.substring(23);
-      const stripeAuthCode = {
-        computedCode,
-      };
-      console.log(stripeAuthCode);
-      setTimeout(
-        () =>
-          axios
-            .post('http://localhost:4000/api/stripe-connect', stripeAuthCode)
-            .then(response => console.log('response'))
-            .catch(err => console.log(err)),
-        2000
-      );
+    // check to see if owner has stripe account connected
+    if (this.props.authTokenRecieved) {
+      axios
+        .get('/api/stripe-connect')
+        .then(response => {
+          console.log('RESPONSE:', response.data);
+          if (response.data.hasStripeID === true) {
+            console.log('SERVER STRIPE ID:', response.data);
+            this.setState({ hasStripeID: true, fetchingStripeID: false });
+          } else {
+            this.setState({ hasStripeID: false, fetchingStripeID: false });
+          }
+          // check to see if auth code is provided
+          if (this.props.location.search) {
+            console.log('BINGO', this.props.location.search.substring(23));
+            let computedCode = this.props.location.search.substring(23);
+            const stripeAuthCode = {
+              computedCode,
+            };
+            console.log(stripeAuthCode);
+
+            axios.post('/api/stripe-connect', stripeAuthCode).then(response => {
+              this.setState({ hasStripeID: true, fetchingStripeID: false });
+            });
+          }
+        })
+        .catch(err => console.log('ERROR CHECKING USER STRIPE ID', err));
     }
   }
 
-  handleChange = event => {
-    this.setState({
-      [event.target.name]: event.target.value,
-      cc: this.state.properties.find(
-        entry => entry.address === event.target.value
-      ).cc,
-      exp: this.state.properties.find(
-        entry => entry.address === event.target.value
-      ).exp,
-      cvv: this.state.properties.find(
-        entry => entry.address === event.target.value
-      ).cvv,
-      history: this.state.properties.find(
-        entry => entry.address === event.target.value
-      ).history,
-    });
-    console.log(
-      this.state.properties.find(entry => entry.address === event.target.value)
-    );
-    console.log(
-      this.state.properties.find(entry => entry.address === event.target.value)
-        .cc
-    );
-    console.log(event.target.value);
-  };
+  componentDidUpdate(prevProps) {
+    if (
+      this.props.authTokenRecieved &&
+      this.props.authTokenRecieved !== prevProps.authTokenRecieved
+    ) {
+      axios
+        .get('/api/stripe-connect')

```

Comment:
Could try extracting this code to its own method and call it during these lifecycle hooks to keep things dry